agenthub/codeact_agent/codeact_agent.py

```diff
@@ -195,6 +195,11 @@ def _get_messages(self, state: State) -> list[Message]:
             ),
         ]
 
+        if 'task' in state.inputs:
```

Comment:
Is this absolutely necessary for CodeAct to be delegated by other agents? cc @li-boxuan

Reply:
I think so, at least last I checked. The task was not accessible to the agent via an event.

I've been thinking we could return it from `get_current_user_intent()` though. It would be easier to use `goal=state.get_user_intent()` for everything, delegate or not.

The issue is, if I recall correctly, that children start with zero history.  The task is included in the DelegateAction, but the child doesn't have access to that action. We'll need Li Boxuan to confirm, though.

Reply:
The contract is messy and there lacks a clear API between agents.

Reply:
It seems like for now the best is to leave it as is, and once the API between agents is defined we can update the agent with a follow-up PR.